crates/shared/src/arguments.rs

```diff
@@ -472,19 +471,16 @@ impl FromStr for ExternalSolver {
 
     fn from_str(solver: &str) -> Result<Self> {
         let parts: Vec<&str> = solver.split('|').collect();
-        ensure!(parts.len() >= 2, "not enough arguments for external solver");
+        ensure!(
+            parts.len() == 2,
```

Comment:
nit: In different places we either use `==` or `>=`. This should probably be aligned.